diff -urNBb --exclude='*.orig' linux-2.6.28.10-org/Documentation/filesystems/00-INDEX linux-2.6.28.10/Documentation/filesystems/00-INDEX
--- linux-2.6.28.10-org/Documentation/filesystems/00-INDEX	2009-05-02 20:54:43.000000000 +0200
+++ linux-2.6.28.10/Documentation/filesystems/00-INDEX	2009-05-05 19:27:11.000000000 +0200
@@ -108,6 +108,8 @@
 	- info on the ufs filesystem.
 vfat.txt
 	- info on using the VFAT filesystem used in Windows NT and Windows 95
+vxext.txt
+	- info on using the VxWorks 5.2+ extended FAT filesystem
 vfs.txt
 	- overview of the Virtual File System
 xfs.txt
diff -urNBb --exclude='*.orig' linux-2.6.28.10-org/fs/Kconfig linux-2.6.28.10/fs/Kconfig
--- linux-2.6.28.10-org/fs/Kconfig	2009-05-02 20:54:43.000000000 +0200
+++ linux-2.6.28.10/fs/Kconfig	2009-05-05 19:27:11.000000000 +0200
@@ -1122,6 +1122,36 @@
 	  Y here.  This will result in _many_ additional debugging messages to be
 	  written to the system log.
 
+config VXEXT_FS
+        tristate "VxWorks extended DOS fs support (EXPERIMENTAL)"
+	depends on EXPERIMENTAL
+	help
+	  The Real-Time Operating System VxWorks v5+ normally ships with an own
+	  FAT16 based DOS filesystem implementation that can work either in a
+	  compatibility mode or in a so-called "extended DOS" (VXEXT1.0) mode.
+	  
+	  While in compatibility mode the normal MSDOS filesystem can be used
+	  and therefore comes with all restrictions like 8+3 and a maximum of
+	  2GB on a partition, running the dosFS part of VxWorks in the
+	  extended version allows to have filenames of up to 40 characters length
+	  and allows to store more than a total of 2GB on a single partition.
+
+	  However, this makes it of course incompatible to the standard FAT16
+	  filesystem as it comes with some specialities for which this separate
+	  implementation is responsible for.
+
+	  So if you want to directly access "extended DOS" VxWorks partitions,
+	  you can select to compile the VXEXT file system support either as a
+	  module (M), or directly into the kernel.
+
+	  Please note that especially write support is highly considered
+	  experimental.
+
+	  So if you haven't heard about all of this before, it's safe to say N.
+
+	  For more information, please consult
+	  <file:Documentation/filesystems/vxext-fs.txt>.
+
 endmenu
 
 menuconfig NETWORK_FILESYSTEMS
diff -urNBb --exclude='*.orig' linux-2.6.28.10-org/fs/Makefile linux-2.6.28.10/fs/Makefile
--- linux-2.6.28.10-org/fs/Makefile	2009-05-02 20:54:43.000000000 +0200
+++ linux-2.6.28.10/fs/Makefile	2009-05-05 19:28:23.000000000 +0200
@@ -81,6 +81,7 @@
 obj-$(CONFIG_CODA_FS)		+= coda/
 obj-$(CONFIG_MINIX_FS)		+= minix/
 obj-$(CONFIG_FAT_FS)		+= fat/
+obj-$(CONFIG_VXEXT_FS)		+= vxext/
 obj-$(CONFIG_BFS_FS)		+= bfs/
 obj-$(CONFIG_ISO9660_FS)	+= isofs/
 obj-$(CONFIG_HFSPLUS_FS)	+= hfsplus/ # Before hfs to find wrapped HFS+
